frontend/src/concepts/connectionTypes/ConnectionTypePreview.tsx

```diff
@@ -1,111 +1,128 @@
 import * as React from 'react';
-import {
-  Button,
-  DescriptionList,
-  DescriptionListDescription,
-  DescriptionListGroup,
-  DescriptionListTerm,
-  Form,
-  FormGroup,
-  FormSection,
-  HelperText,
-  HelperTextItem,
-  LabelGroup,
-  MenuToggleStatus,
-  Popover,
-  Title,
-} from '@patternfly/react-core';
+import { Form, FormGroup, FormSection, MenuToggleStatus, Title } from '@patternfly/react-core';
 import ConnectionTypeFormFields from '~/concepts/connectionTypes/fields/ConnectionTypeFormFields';
-import { ConnectionTypeConfigMapObj } from '~/concepts/connectionTypes/types';
-import NameDescriptionField from '~/concepts/k8s/NameDescriptionField';
-import { getDescriptionFromK8sResource } from '~/concepts/k8s/utils';
-import UnspecifiedValue from '~/concepts/connectionTypes/fields/UnspecifiedValue';
-import SimpleSelect from '~/components/SimpleSelect';
-import CategoryLabel from '~/concepts/connectionTypes/CategoryLabel';
-import TruncatedText from '~/components/TruncatedText';
+import {
+  ConnectionTypeConfigMapObj,
+  ConnectionTypeValueType,
+} from '~/concepts/connectionTypes/types';
+import {
+  getDisplayNameFromK8sResource,
+  getResourceNameFromK8sResource,
+} from '~/concepts/k8s/utils';
+import TypeaheadSelect, { TypeaheadSelectOption } from '~/components/TypeaheadSelect';
+import K8sNameDescriptionField from '~/concepts/k8s/K8sNameDescriptionField/K8sNameDescriptionField';
+import {
+  K8sNameDescriptionFieldData,
+  K8sNameDescriptionFieldUpdateFunction,
+} from '~/concepts/k8s/K8sNameDescriptionField/types';
+import { ConnectionTypeDetailsHelperText } from './ConnectionTypeDetailsHelperText';
 
 type Props = {
   obj?: ConnectionTypeConfigMapObj;
+  setObj?: (obj?: ConnectionTypeConfigMapObj) => void;
+  connectionTypes?: ConnectionTypeConfigMapObj[];
+  isPreview?: boolean;
+  connectionNameDesc?: K8sNameDescriptionFieldData;
+  setConnectionNameDesc?: K8sNameDescriptionFieldUpdateFunction;
+  connectionValues?: {
+    [key: string]: ConnectionTypeValueType;
+  };
+  setConnectionValues?: (values: { [key: string]: ConnectionTypeValueType }) => void;
 };
 
-// TODO consider refactoring this form for reuse when creating connection type instances
-const ConnectionTypePreview: React.FC<Props> = ({ obj }) => {
+const ConnectionTypePreview: React.FC<Props> = ({
+  obj,
+  setObj,
+  connectionTypes,
+  isPreview,
+  connectionNameDesc,
+  setConnectionNameDesc,
+  connectionValues,
+  setConnectionValues,
+}) => {
   const connectionTypeName = obj && obj.metadata.annotations?.['openshift.io/display-name'];
```

Comment:
I think if this is not a preview, we want to fall back to the metadata name therefore using the util `getDisplayNameFromK8Resource`